### ISSUE TITLE
Pyright fails to properly narrow the type of a constrained `TypeVar` in some cases when one of the constrained types is `None`

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#1840]
Pyright fails to properly narrow the type of a constrained `TypeVar` when:
- The `TypeVar` includes `None` as one of the constrained types.
- The constrained variable is narrowed with a `None` check, eg. `var is not None`

Code sample:
```py
from typing import TypeVar

_T = TypeVar("_T", None, str)


def func(val: _T) -> _T:
    if val is not None:
        reveal_type(val)  # Type of "val" is "_T@func"
        return val
    else:
        reveal_type(val)  # Type of "val" is "Never"
        return val
        # Expression of type "Never" cannot be assigned to return type "_T@func"
            # Type "Never" cannot be assigned to type "_T@func"
```

I would expect pyright to properly narrow the type as it does without issue in this example:

```py
from typing import TypeVar

_T = TypeVar("_T", None, str)


def func(val: _T) -> _T:
    if isinstance(val, str):
        reveal_type(val)  # Type of "val" is "str"
        return val
    else:
        reveal_type(val)  # Type of "val" is "None"
        return val
```

Mypy correctly narrows the type in both cases.

Tested on pyright `1.1.138` and mypy `0.812`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.139, which I've just published. It will also be included in the next release of pylance.
